metrics.go

```diff
@@ -0,0 +1,434 @@
+	}
+
+	return sdm, nil
+}
+
+func metricLableKeysToLabels(defaults map[string]labelValue, labelKeys []string) []*labelpb.LabelDescriptor {
+	labelDescriptors := make([]*labelpb.LabelDescriptor, 0, len(defaults)+len(labelKeys))
+
+	// Fill in the defaults first.
+	for key, lbl := range defaults {
+		labelDescriptors = append(labelDescriptors, &labelpb.LabelDescriptor{
+			Key:         sanitize(key),
+			Description: lbl.desc,
+			ValueType:   labelpb.LabelDescriptor_STRING,
+		})
+	}
+
+	// Now fill in those from the metric.
+	for _, key := range labelKeys {
+		labelDescriptors = append(labelDescriptors, &labelpb.LabelDescriptor{
+			Key:         sanitize(key),
+			Description: "",                             // TODO: [rghetia] when descriptor is available use that.
+			ValueType:   labelpb.LabelDescriptor_STRING, // We only use string tags
+		})
+	}
+	return labelDescriptors
+}
+
+func metricDescriptorTypeToMetricKind(m *metricdata.Metric) (googlemetricpb.MetricDescriptor_MetricKind, googlemetricpb.MetricDescriptor_ValueType) {
+	if m == nil {
+		return googlemetricpb.MetricDescriptor_METRIC_KIND_UNSPECIFIED, googlemetricpb.MetricDescriptor_VALUE_TYPE_UNSPECIFIED
+	}
+
+	switch m.Descriptor.Type {
+	case metricdata.TypeCumulativeInt64:
+		return googlemetricpb.MetricDescriptor_CUMULATIVE, googlemetricpb.MetricDescriptor_INT64
+
+	case metricdata.TypeCumulativeFloat64:
+		return googlemetricpb.MetricDescriptor_CUMULATIVE, googlemetricpb.MetricDescriptor_DOUBLE
+
+	case metricdata.TypeCumulativeDistribution:
+		return googlemetricpb.MetricDescriptor_CUMULATIVE, googlemetricpb.MetricDescriptor_DISTRIBUTION
+
+	case metricdata.TypeGaugeFloat64:
+		return googlemetricpb.MetricDescriptor_GAUGE, googlemetricpb.MetricDescriptor_DOUBLE
+
+	case metricdata.TypeGaugeInt64:
+		return googlemetricpb.MetricDescriptor_GAUGE, googlemetricpb.MetricDescriptor_INT64
+
+	case metricdata.TypeGaugeDistribution:
+		return googlemetricpb.MetricDescriptor_GAUGE, googlemetricpb.MetricDescriptor_DISTRIBUTION
+
+	case metricdata.TypeSummary:
+		// TODO: [rghetia] after upgrading to proto version3, retrun UNRECOGNIZED instead of UNSPECIFIED
+		return googlemetricpb.MetricDescriptor_METRIC_KIND_UNSPECIFIED, googlemetricpb.MetricDescriptor_VALUE_TYPE_UNSPECIFIED
+
+	default:
+		// TODO: [rghetia] after upgrading to proto version3, retrun UNRECOGNIZED instead of UNSPECIFIED
+		return googlemetricpb.MetricDescriptor_METRIC_KIND_UNSPECIFIED, googlemetricpb.MetricDescriptor_VALUE_TYPE_UNSPECIFIED
+	}
+}
+
+func metricRscToMpbRsc(rs *resource.Resource) *monitoredrespb.MonitoredResource {
+	if rs == nil {
+		return &monitoredrespb.MonitoredResource{
+			Type: "global",
+		}
+	}
+	typ := rs.Type
+	if typ == "" {
+		typ = "global"
+	}
+	mrsp := &monitoredrespb.MonitoredResource{
+		Type: typ,
+	}
+	if rs.Labels != nil {
+		mrsp.Labels = make(map[string]string, len(rs.Labels))
+		for k, v := range rs.Labels {
```

Comment:
We may need to do some mappings from OC resource labels to SD resource labels after https://github.com/census-instrumentation/opencensus-specs/pull/235. You can leave a TODO here for now.

Reply:
added a TODO